src/main/java/htsjdk/samtools/util/IntervalList.java

```diff
@@ -379,13 +379,14 @@ public void unique(final boolean concatenateNames) {
     }
 
     /**
-     * Merges a sorted collection of intervals and optionally concatenates unique names or takes the first name.
+     * Merges a collection of intervals and optionally concatenates unique names or takes the first name.
+     * Uses the contig and strand of the first interval seen without validating that the others match.
      */
     static Interval merge(final Iterable<Interval> intervals, final boolean concatenateNames) {
         final Interval first = intervals.iterator().next();
         final String chrom = first.getContig();
         int start = first.getStart();
-        int end = start;
+        int end = first.getEnd();
```

Comment:
nice catch. sorry.

Reply:
I definitely didn't notice it in review.  It makes sense, you can't have an end before start..